# HG changeset patch
# User Andreas Weber <andy.weber.aw@gmail.com>
# Date 1421065785 -3600
#      Mon Jan 12 13:29:45 2015 +0100
# Node ID 810cfb00b72ba2bc694ba3f8cabdb9a65d94f6a7
# Parent  c304687571c8c59f84a3343d7dc060b9cba319c5
portaudio_play_callback: Fix stereo playback (bug #43965)

diff --git a/libinterp/dldfcn/audiodevinfo.cc b/libinterp/dldfcn/audiodevinfo.cc
--- a/libinterp/dldfcn/audiodevinfo.cc
+++ b/libinterp/dldfcn/audiodevinfo.cc
@@ -649,17 +649,17 @@ portaudio_play_callback (const void *, v
   // a second copy of the *entire* data array to be made when only a
   // small portion (BUFFER_SIZE elements) is usually needed for this
   // callback.
 
   const RowVector sound_l = player->get_left ();
   const RowVector sound_r = player->get_right ();
 
   const double *pl = sound_l.data ();
-  const double *pr = sound_l.data ();
+  const double *pr = sound_r.data ();
 
   if (player->get_type () == TYPE_DOUBLE)
     {
       switch (player->get_nbits ())
         {
         case 8:
           {
             static double scale_factor = std::pow (2.0, 7) - 1.0;
